src/devices/PiJuice/PiJuiceStatus.cs

```diff
@@ -0,0 +1,262 @@
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.ButtonEvent, 2);
+
+            var buttonEvents = new List<ButtonEventType>(3)
+            {
+                Enum.IsDefined(typeof(ButtonEventType), response[0] & 0x0F) ? (ButtonEventType)(response[0] & 0x0F) : ButtonEventType.Unknown,
+                Enum.IsDefined(typeof(ButtonEventType), (response[0] >> 4) & 0x0F) ? (ButtonEventType)((response[0] >> 4) & 0x0F) : ButtonEventType.Unknown,
+                Enum.IsDefined(typeof(ButtonEventType), response[1] & 0x0F) ? (ButtonEventType)(response[1] & 0x0F) : ButtonEventType.Unknown,
+            };
+
+            return buttonEvents;
+        }
+
+        /// <summary>
+        /// Clears generated button event
+        /// </summary>
+        /// <param name="button">Button to clear button event for</param>
+        public void ClearButtonEvent(Button button)
+        {
+            var array = button switch
+            {
+                Button.Switch1 => new byte[] { 0xF0, 0xFF },
+                Button.Switch2 => new byte[] { 0x0F, 0xFF },
+                Button.Switch3 => new byte[] { 0xFF, 0xF0 },
+                _ => throw new NotImplementedException()
+            };
+
+            _piJuice.WriteCommand(PiJuiceCommand.ButtonEvent, array);
+        }
+
+        /// <summary>
+        /// Get battery charge level between 0 and 100 percent
+        /// </summary>
+        /// <returns>Battery charge level percentage</returns>
+        public byte GetChargeLevel()
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.ChargeLevel, 1);
+
+            return response[0];
+        }
+
+        /// <summary>
+        /// Get battery temperature
+        /// </summary>
+        /// <returns>Battery temperature in celsius</returns>
+        public Temperature GetBatteryTemperature()
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.BatteryTemperature, 2);
+
+            return new Temperature(BinaryPrimitives.ReadInt16LittleEndian(response), TemperatureUnit.DegreeCelsius);
+        }
+
+        /// <summary>
+        /// Get battery voltage
+        /// </summary>
+        /// <returns>Battery voltage in millivolts</returns>
+        public ElectricPotential GetBatteryVoltage()
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.BatteryVoltage, 2);
+
+            return new ElectricPotential(BinaryPrimitives.ReadInt16LittleEndian(response), ElectricPotentialUnit.Millivolt);
+        }
+
+        /// <summary>
+        /// Get battery current
+        /// </summary>
+        /// <returns>Battery current in milliamps</returns>
+        public ElectricCurrent GetBatteryCurrent()
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.BatteryCurrent, 2);
+
+            int i = BinaryPrimitives.ReadInt16LittleEndian(response);
+            if ((i & (1 << 15)) == (1 << 15))
+            {
+                i -= (1 << 16);
+            }
```

Comment:
Hi,
I'm just another .NET developer (unrelated to this project), and had a need for exactly this sort of library (saving me a bunch of work, thanks! 👍)
I've found when testing that ReadInt16LittleEndian appears to already have done the conversion that's being done here (resulting in a value 1<<16 less than expected).

```suggestion
```

This also exists in the GetIOCurrent() function a little bit lower down.

Thanks!

Reply:
@jstoker can you file an issue on that? also if you have the device to test I'd recommend to just send a PR with the fix

Reply:
I've created https://github.com/dotnet/iot/issues/1333